django/laws/views.py

```diff
@@ -372,19 +324,23 @@ def htmx_sse_response(response_gen):
             message_html = llm_response_to_html(full_message)
             message_html_lines = message_html.split("\n")
 
+        sync_to_async(llm.create_costs)()
+        sync_to_async(cache.delete)(f"sources_{query}")
+
         yield (
             f"data: <div hx-swap-oob='true' id='answer-sse'>"
             f"<div>{sse_joiner.join(message_html_lines)}</div></div>\n\n"
```

Comment:
Sometimes the AI answer gets streamed out, then disappears and starts over, then completes. Not sure why. This problem may predate this PR.